collated/main/jelneo.md

```diff
@@ -0,0 +1,1244 @@
+``` java
+/**
+ * Tests that a {@code Person}'s {@code Tag} matches any of the keywords given.
+ */
+public class PersonContainsTagPredicate implements Predicate<ReadOnlyPerson> {
+    private final List<String> tagKeywords;
+
+    public PersonContainsTagPredicate(List<String> tagKeywords) {
+        this.tagKeywords = tagKeywords;
+    }
+
+    /**
+     * Evaluates this predicate on the given person.
+     *
+     * @return {@code true} if the person matches the predicate,
+     * otherwise {@code false}
+     */
+    @Override
+    public boolean test(ReadOnlyPerson person) {
+        Set<Tag> tagList = person.getTags();
+        for (Tag tag : tagList) {
+            if (tagKeywords.stream().anyMatch(keyword -> StringUtil.containsWordIgnoreCase(tag.tagName, keyword))) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof NameContainsKeywordsPredicate // instanceof handles nulls
+                && this.tagKeywords.equals(((PersonContainsTagPredicate) other).tagKeywords)); // state check
+    }
+}
+```
+###### \java\seedu\address\ui\CommandBox.java
+``` java
+
+    /**
+     * Masks password starting from second whitespace(if it exists) until the end of input
+     */
+    private void handlePasswordMasking() {
+        String currentInput = commandTextField.getText();
+        numOfSpaces = getNumOfSpaces(currentInput);
+        prevInputLength = inputLength;
+        inputLength = commandTextField.getLength();
+        indexOfFirstWhitespace = currentInput.indexOf(" ");
+        currentMaskFromIndex = currentInput.indexOf(" ", indexOfFirstWhitespace + 1) + 1;
+
+        // update index indicating where to start masking from if user deletes and re-enters password
+        if (maskFromIndex == 0) {
+            maskFromIndex = currentMaskFromIndex;
+        }
+        // handles the case where user backspaces and change password input from current caret location to
+        // before second whitespace
+        if (!passwordFromInput.isEmpty() && inputLength < prevInputLength) {
+            handleBackspaceEvent();
+        }
+        // user's caret is before the second whitespace backspace or after second whitespace
+        // initialise fields if password is keyed in then deleted
+        if (passwordFromInput.isEmpty() && prevInputLength != 0) {
+            initialiseVariablesUsedInMasking();
+        }
+        // starts masking password after the second whitespace and prevent the reading of the asterisk after replacing
+        // a character in the command box text field with an asterisk
+        if (numOfSpaces >= 2 && currentInput.charAt(currentInput.length() - 1) != ' '
+                && currentInput.charAt(currentInput.length() - 1) != BLACK_CIRCLE) {
+            maskPasswordInput(currentInput);
+
```

Comment:
Use blank lines sparingly. It's good to have blank lines to separate logical code blocks, but too many blank lines will increase the height of the code, which increases the amount of scrolling.